recipes/libheif/all/conanfile.py

```diff
@@ -76,10 +79,17 @@ def _configure_cmake(self):
         self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_X265"] = not self.options.with_x265
         self._cmake.definitions["WITH_X265"] = self.options.with_x265
         # aom
-        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_LibAOM"] = not self.options.get_safe("with_libaomav1", False)
-        self._cmake.definitions["WITH_AOM"] = self.options.get_safe("with_libaomav1", False)
+        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_LibAOM"] = not self.options.with_libaomav1
+        self._cmake.definitions["WITH_AOM"] = self.options.with_libaomav1
         # dav1d
         self._cmake.definitions["WITH_DAV1D"] = self.options.get_safe("with_dav1d", False)
+
+        # Workaround for cross-build to at least iOS/tvOS/watchOS,
+        # when dependencies are found with find_path() and find_library()
+        if tools.cross_building(self):
+            self._cmake.definitions["CMAKE_FIND_ROOT_PATH_MODE_INCLUDE"] = "BOTH"
+            self._cmake.definitions["CMAKE_FIND_ROOT_PATH_MODE_LIBRARY"] = "BOTH"
```

Comment:
Is this because `CMAKE_FIND_ROOT_PATH` is not properly set?

Reply:
Yes and No. I think that `CMAKE_FIND_ROOT_PATH` should not be modified by conan, or at least conan should not inject root path of packages in `CMAKE_FIND_ROOT_PATH`, because it may append executables of host context in top priority for `find_program()`.

Reply:
That's why `CMAKE_FIND_ROOT_PATH_MODE_PROGRAM` should be set to `NEVER` (when cross building)

Reply:
But `CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER` is quite strong, and might break consumers who want to inject their own paths in `CMAKE_FIND_ROOT_PATH`.

Reply:
Those people are abusing a system for something it isn't built for.
Take mingw@Linux as an example.

The executables for the build machine are stored in `/usr/bin`.
The build machines root is `/usr`.

For the host (=Windows), the root is `/usr/i686-w64-mingw32/sys-root/mingw/`.

So for cross building on Linux, you need add the following to your cmake toolchain:
```cmake
set(CMAKE_FIND_ROOT_PATH "/usr/i686-w64-mingw32/sys-root/mingw/")

set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)  # Only look in CMAKE_FIND_ROOT_PATH
set(CMAKE_FIND_ROOT_PATH_MODE_PATH ONLY)  # Only look in CMAKE_FIND_ROOT_PATH
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)  # Only look in CMAKE_FIND_ROOT_PATH
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)  # NEVER look in CMAKE_FIND_ROOT_PATH
```

Reply:
@madebr could you share your suggestion in https://github.com/conan-io/conan/issues/9202 please?

There are many way to fix this issue, and I don't know what it the best one:

- force all `CMAKE_FIND_ROOT_PATH_MODE_*` to BOTH (or NEVER)
- force all `CMAKE_FIND_ROOT_PATH_MODE_PROGRAM` to NEVER, leave others `CMAKE_FIND_ROOT_PATH_MODE_*` untouched, and prepend root packages paths as well as `CMAKE_CURRENT_LIST_DIR` in `CMAKE_FIND_ROOT_PATH`.